Add `ID` field for `docker plugin ls`

yongtang opened the pull request:
**- What I did**

This fix tries to address the enhancement proposed in #28708 to display ID field for the output of `docker plugin ls`.

**- How I did it**

This fix add `ID` field to the output of `docker plugin ls`

Related docs has been updated.

**- How to verify it**

The following is the sample output:
```

ubuntu@ubuntu:~/docker$ docker plugin ls
ID                  NAME                  TAG                 DESCRIPTION                ENABLED
69553ca1d123        tiborvass/no-remove   latest              A test plugin for Docker   true
ubuntu@ubuntu:~/docker$ docker plugin ls --no-trunc
ID                                                                 NAME                  TAG                 DESCRIPTION                ENABLED
69553ca1d1231d7f65b67a2bad8dcb4be56c6b6c6125d16c7f28cd554e6e3c1d   tiborvass/no-remove   latest              A test plugin for Docker   true

```

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

![large](https://cloud.githubusercontent.com/assets/6932348/20548817/9a23f474-b0db-11e6-8f90-d41a8caa9498.jpg)


This fix fixes #28708.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

allencloud commented:
Oh, that is really great for PLUGIN. Thanks a lot @yongtang 

I think this is helpful for client side to get IDs for every plugin. And it is indeed a client side work.  While this fixed one part of the issue. Since we still cannot manage plugin via IDs, since maybe we need to change daemon side as well. Currently, the plugin API can only accept plugin name, no plugin ID supported. Then we still cannot use command `docker plugin inspect PLUGIN_ID`.

yongtang commented:
Thanks for the feedback @allencloud. I created another PR #28789 so that `docker plugin inspect` is able to search based on name or ID (or ID prefix in case it is the short form). Please take a look.

thaJeztah commented on cliconfig/configfile/file.go:
```diff
@@ -27,6 +27,7 @@ type ConfigFile struct {
 	PsFormat             string                      `json:"psFormat,omitempty"`
 	ImagesFormat         string                      `json:"imagesFormat,omitempty"`
 	NetworksFormat       string                      `json:"networksFormat,omitempty"`
+	PluginsFormat        string                      `json:"pluginsFormat,omitempty"`
```
Don't forget to add an example to the example configuration file here; https://github.com/docker/docker/blob/master/docs/reference/commandline/cli.md#configuration-files

yongtang replied:
Thanks @thaJeztah. The related docs has been updated.

thaJeztah commented:
Design LGTM, I think this is consistent with other overviews we have, and (if filtering is added at some point), allows things such as `docker plugin rm $(docker plugin ls -q --filter status=disabled)`

thaJeztah commented:
ping @tiborvass @anusha-ragunathan @vieux PTAL

anusha-ragunathan commented:
@yongtang : Thanks for working on this. Listing by pluginID is necessary for `docker plugin inspect pluginID` to work seamlessly. So we need this for 1.13. 

But this PR has formatting changes which are not necessary for 1.13. So do you mind splitting this into 2 PRs (list-ID + formatting changes), so that we can merge the list-ID PR for 1.13 and merge the other one on master?

yongtang committed:
Add `ID` field for `docker plugin ls`

This fix tries to address the enhancement proposed in 28708 to display
ID field for the output of `docker plugin ls`.

This fix add `ID` field to the output of `docker plugin ls`

Related docs has been updated.

This fix fixes 28708.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang commented:
Thanks @anusha-ragunathan. I splitter this PR into two.

Now this PR only adds `ID` fields to `docker plugin ls`. 

Another PR #29088 adds `--format` and `--quiet` flag to `docker plugin ls`.

Please take a look and let me know if there are any issues.

anusha-ragunathan commented:
LGTM (granted tests pass)

thaJeztah reviewed (approved):
LGTM
